mitodl/odl-video-service
# Email templates

#### What are the relevant tickets?
Closes https://trello.com/c/ch9rjBlN/28-update-notification-email-when-video-is-ready

#### What's this PR do?
- Adds HTML email templates to OVS
- Does not yet remove the `NotificationEmail` model - my thought is that it should be removed in a subsequent PR in case this one needs to be reverted.

#### How should this be manually tested?
- Set `MAILGUN_RECIPIENT_OVERRIDE=<your_email>` and other MAILGUN .env variables (you can get them from CI)
- Set `ODL_VIDEO_FROM_EMAIL=MIT ODL Video <odl-video-support@mit.edu>`
- Set `ODL_VIDEO_SUPPORT_EMAIL=MIT ODL Video <odl-video-support@mit.edu>`
- Run the following, you should get 3 emails afterward:
```
from mail.tasks import *
from ui.models import *
video = Video.objects.order_by("-created_at").first()
for status in [VideoStatus.COMPLETE, VideoStatus.TRANSCODE_FAILED_VIDEO, VideoStatus.UPLOAD_FAILED]:
    video.status = status
    send_notification_email(video)
```


#### Screenshots (if appropriate)

SUCCESS:

<img width="1004" alt="Screen Shot 2020-04-23 at 1 43 21 PM" src="https://user-images.githubusercontent.com/187676/80131657-8dc7dc80-8568-11ea-99d2-79587ef50d14.png">



ERROR LIKELY DUE TO INVALID FORMAT:

<img width="1009" alt="Screen Shot 2020-04-23 at 1 43 36 PM" src="https://user-images.githubusercontent.com/187676/80131643-87d1fb80-8568-11ea-9d43-c023bcc6d4b5.png">



OTHER ERRORS:

<img width="721" alt="Screen Shot 2020-04-23 at 1 24 09 PM" src="https://user-images.githubusercontent.com/187676/80131085-b1d6ee00-8567-11ea-85a2-824f78d9013b.png">
